client/src/main/scala/com.gu.contentapi.client/ContentApiClient.scala

```diff
@@ -44,9 +50,24 @@ trait ContentApiClient {
   /** Authentication and format parameters appended to each query */
   private def parameters = Map("api-key" -> apiKey, "format" -> "thrift")
 
+  private lazy val backoffStrategy = Backoff(Duration(250, TimeUnit.MILLISECONDS))
+
   /** Streamlines the handling of a valid CAPI response */
-  private def fetchResponse(contentApiQuery: ContentApiQuery)(implicit context: ExecutionContext): Future[Array[Byte]] =
-    get(url(contentApiQuery), headers).flatMap(HttpResponse.check)
+  private def fetchResponse(contentApiQuery: ContentApiQuery)(implicit context: ExecutionContext): Future[Array[Byte]] = {
+    def getter: Future[HttpResponse] = get(url(contentApiQuery), headers)(context)
+
+    getter
+      .recoverWith {
+        case r: ContentApiRecoverableException =>
+          println(s"Recoverable error ${r.httpStatus} encountered - retrying")
+          backoffStrategy.retry(getter)
+
+        case e => Future.failed(e)
+      }
+      .map {
+        HttpResponse.check
+      }
+  }.flatten
```

Comment:
There are still a couple of `.flatten`s knocking around in these changes, suggesting there's still something up with the inner workings of the `ScheduledExecutor`. I'd like to clear that up.